src/pubsub/message/rpc.d.ts

```diff
@@ -0,0 +1,222 @@
+
+    /**
+     * Creates a RPC message from a plain object. Also converts values to their respective internal types.
+     * @param d Plain object
+     * @returns RPC
+     */
+    public static fromObject(d: { [k: string]: any }): RPC;
+
+    /**
+     * Creates a plain object from a RPC message. Also converts values to other types if specified.
+     * @param m RPC
+     * @param [o] Conversion options
+     * @returns Plain object
+     */
+    public static toObject(m: RPC, o?: $protobuf.IConversionOptions): { [k: string]: any };
+
+    /**
+     * Converts this RPC to JSON.
+     * @returns JSON object
+     */
+    public toJSON(): { [k: string]: any };
+}
+
+export namespace RPC {
+
+    /** Properties of a SubOpts. */
+    interface ISubOpts {
+
+        /** SubOpts subscribe */
+        subscribe?: (boolean|null);
+
+        /** SubOpts topicID */
+        topicID?: (string|null);
+    }
+
+    /** Represents a SubOpts. */
+    class SubOpts implements ISubOpts {
+
+        /**
+         * Constructs a new SubOpts.
+         * @param [p] Properties to set
+         */
+        constructor(p?: RPC.ISubOpts);
+
+        /** SubOpts subscribe. */
+        public subscribe: boolean;
+
+        /** SubOpts topicID. */
+        public topicID: string;
+
+        /**
+         * Encodes the specified SubOpts message. Does not implicitly {@link RPC.SubOpts.verify|verify} messages.
+         * @param m SubOpts message or plain object to encode
+         * @param [w] Writer to encode to
+         * @returns Writer
+         */
+        public static encode(m: RPC.ISubOpts, w?: $protobuf.Writer): $protobuf.Writer;
+
+        /**
+         * Decodes a SubOpts message from the specified reader or buffer.
+         * @param r Reader or buffer to decode from
+         * @param [l] Message length if known beforehand
+         * @returns SubOpts
+         * @throws {Error} If the payload is not a reader or valid buffer
+         * @throws {$protobuf.util.ProtocolError} If required fields are missing
+         */
+        public static decode(r: ($protobuf.Reader|Uint8Array), l?: number): RPC.SubOpts;
+
+        /**
+         * Creates a SubOpts message from a plain object. Also converts values to their respective internal types.
+         * @param d Plain object
+         * @returns SubOpts
+         */
+        public static fromObject(d: { [k: string]: any }): RPC.SubOpts;
+
+        /**
+         * Creates a plain object from a SubOpts message. Also converts values to other types if specified.
+         * @param m SubOpts
+         * @param [o] Conversion options
+         * @returns Plain object
+         */
+        public static toObject(m: RPC.SubOpts, o?: $protobuf.IConversionOptions): { [k: string]: any };
+
+        /**
+         * Converts this SubOpts to JSON.
+         * @returns JSON object
+         */
+        public toJSON(): { [k: string]: any };
+    }
+
+    /** Properties of a Message. */
+    interface IMessage {
```

Comment:
why are the properties here all both optional and allow `null`?

Reply:
This is generated from the protobuf [spec](https://github.com/libp2p/specs/tree/master/pubsub#the-message) for pubsub where no part of the message is required. But basically, "Depending on the signature policy and message ID function."